src/main/java/com/flightstats/hub/cluster/ClusterCacheDao.java

```diff
@@ -163,20 +157,28 @@ public void delete(String name, String basePath) {
         }
     }
 
-    private LastUpdated getLastUpdated(String path) throws Exception {
+    private MostRecentData getMostRecentData(String path) throws Exception {
         Stat stat = new Stat();
         byte[] bytes = curator.getData().storingStatIn(stat).forPath(path);
         Optional<ContentPath> pathOptional = ContentPath.fromUrl(new String(bytes, StandardCharsets.UTF_8));
-        return new LastUpdated(pathOptional.get(), stat.getVersion());
+        return new MostRecentData(pathOptional.get(), stat.getVersion());
     }
 
-    private class LastUpdated {
-        ContentPath key;
-        int version;
+    private class MostRecentData {
```

Comment:
'Data' is almost always a bad choice for a name in a class (or any name, barring something like raw bytes from a file). Everything is data. Clean code calls them "noise words", words that really don't tell you any more than if they were left off. It specifically calls out 'data' and 'info' as common culprits. 

MostRecentPathVersion? MostRecentKey?

Reply:
Yeah, for sure.  VersionedKey, since that's what it is.  
Interesting that we never do anything with versions in hub though...wonder if we could just kill that...

Reply:
The only versions I can think of that might be related to the hub are S3 versioning (which I don't think the hub knows anything about) and the actual hub version that's deployed, which data shouldn't care about and isn't an integer. If you know offhand, what is this one supposed to mean?

Reply:
Hmm...well...that's an interesting read.  ZNodes have an inherent version.  ZK gives us a node at version 11 and expects us to pass that back in to say, "we're making an update relative to version 11".  That way if there's a version 12 ZK can reject the update with a KeeperException.BadVersionException.
We catch that and retry if our condition is still true (newer | older).  So...definitely want to keep that around.  :)